client/rest-high-level/src/test/java/org/elasticsearch/client/RequestConvertersTests.java

```diff
@@ -1229,7 +1235,103 @@ public void testRenderSearchTemplate() throws Exception {
+    
+            // scroll is not supported in the current msearch or msearchtemplate api, so unset it:
+            searchRequest.scroll((Scroll) null);
+            // batched reduce size is currently not set-able on a per-request basis as it is a query string parameter only
+            searchRequest.setBatchedReduceSize(SearchRequest.DEFAULT_BATCHED_REDUCE_SIZE);
+            
+            setRandomIndicesOptions(searchRequest::indicesOptions, searchRequest::indicesOptions, expectedParams);
+            
+            SearchTemplateRequest searchTemplateRequest = new SearchTemplateRequest(searchRequest);
+    
+            searchTemplateRequest.setScript("{\"query\": { \"match\" : { \"{{field}}\" : \"{{value}}\" }}}");
+            searchTemplateRequest.setScriptType(ScriptType.INLINE);
+            searchTemplateRequest.setProfile(randomBoolean());
+    
+            Map<String, Object> scriptParams = new HashMap<>();
+            scriptParams.put("field", "name");
+            scriptParams.put("value", randomAlphaOfLengthBetween(2, 5));
+            searchTemplateRequest.setScriptParams(scriptParams);
+    
+            multiSearchTemplateRequest.add(searchTemplateRequest);            
+        }
+
+        Request multiRequest = RequestConverters.multiSearchTemplate(multiSearchTemplateRequest);
+        
+        assertEquals(HttpPost.METHOD_NAME, multiRequest.getMethod());
+        assertEquals("/_msearch/template", multiRequest.getEndpoint());
+        List<SearchTemplateRequest> searchRequests = multiSearchTemplateRequest.requests();
+        assertEquals(numSearchRequests, searchRequests.size());
+
+        //Serialize the request
+        byte[] bytes = MultiSearchTemplateRequest.writeMultiLineFormat(multiSearchTemplateRequest, XContentType.JSON.xContent());
+        String jsonRequestString = new String(bytes, StandardCharsets.UTF_8);
+        // Deserialize the request
+        TestRestRequest tr = new TestRestRequest(multiRequest.getEndpoint(), jsonRequestString, XContentType.JSON);
+        MultiSearchTemplateRequest deser = RestMultiSearchTemplateAction.parseRequest(tr, true);
+
+        // For object equality purposes need to set the search requests' source to non-null
+        for (SearchTemplateRequest str : deser.requests()) {
+            SearchRequest sr = str.getRequest();
+            if (sr.source() == null) {
+                sr.source(new SearchSourceBuilder());
+            }
+        }        
+        // Compare the deserialized request object with the original request 
+        assertEquals(multiSearchTemplateRequest, deser);
+        
+        // Finally, serialize the deserialized request to compare JSON equivalence (in case Object.equals() fails to reveal a discrepancy)
+        String reserJsonRequestString = new String(MultiSearchTemplateRequest.writeMultiLineFormat(deser, XContentType.JSON.xContent()),
+                StandardCharsets.UTF_8);
+        assertEquals(jsonRequestString, reserJsonRequestString);
```

Comment:
I think that the multiline format should be tested in elasticsearch given that the code belongs there? Something along the lines of `MultiSearchRequestTests#testMultiLineSerialization`?

Reply:
> I think that the multiline format should be tested in elasticsearch given that the code belongs there?

Perhaps not core - MultiSearchTemplateRequest is in a module because of the mustache dependency. Maybe a new class in /:modules:lang-mustache/src/test/java/org.elasticsearch.script.mustache ?

Reply:
yea I mean where the request sits.

Reply:
to clarify, I would leave the initial part and test here only the result of `RequestConverters#multiSearchTemplate`, and would move the multi line serialization roundtrips to another test to be placed where the request lives.